lib/mfx/api.py

```diff
@@ -0,0 +1,141 @@
+from ...lib.codecs import Codec
+from ...lib.common import memoize
+from ...lib.properties import PropertyHandler
+
+# TODO: move string_api enums to this file
+# NOTE/FIXME: these enums need to derive from IntEnum, not Enum
+from ...lib.string_api import CodecProfile as MfxCodecProfile
+from ...lib.string_api import CodecLevel as MfxCodecLevel
+from ...lib.string_api import RateControlMethod as MfxRateControlMethod
+
+class StringAPI(PropertyHandler):
+  level   = property(lambda s: s.ifprop("level", "CodecLevel={level}"))
+  gop     = property(lambda s: s.ifprop("gop", "GopPicSize={gop}"))
+  bframes = property(lambda s: s.ifprop("bframes", "GopRefDist={bframes}"))
+  slices  = property(lambda s: s.ifprop("slices", "NumSlice={slices}"))
+  minrate = property(lambda s: s.ifprop("minrate", "TargetKbps={minrate}"))
+  maxrate = property(lambda s: s.ifprop("maxrate", "MaxKbps={maxrate}"))
+  refs    = property(lambda s: s.ifprop("refs", "NumRefFrame={refs}"))
+  ladepth = property(lambda s: s.ifprop("ladepth", "LookAheadDepth={ladepth}"))
+  extbrc  = property(lambda s: s.ifprop("extbrc", "ExtBRC={extbrc}"))
+  ldb     = property(lambda s: s.ifprop("ldb", "mfxExtCodingOption3.LowDelayBRC={ldb}"))
+  strict  = property(lambda s: s.ifprop("strict", "GopOptFlag={strict}"))
+  pict    = property(lambda s: s.ifprop("vpict", "mfxExtCodingOption.PicTimingSEI=0"))
+  quality = property(lambda s: s.ifprop("quality", "TargetUsage={quality}"))
+  lowpower  = property(lambda s: s.ifprop("lowpower", "LowPower={lowpower}"))
+  tilecols  = property(lambda s: s.ifprop("tilecols", "mfxExtAV1TileParam.NumTileColumns={tilecols}"))
+  tilerows  = property(lambda s: s.ifprop("tilerows", "mfxExtAV1TileParam.NumTileRows={tilerows}"))
+  maxframesize  = property(lambda s: s.ifprop("maxframesize", "MaxSliceSize={maxframesize}"))
+
+  @property
+  def profile(self):
+    def inner(profile):
+      return f"CodecProfile={self.map_profile(self.codec, self.props['profile'])}"
+    return self.ifprop("profile", inner)
+
+  @property
+  def rcmode(self):
+    return f"RateControlMethod={self.map_rcmode(self.props['rcmode'])}"
+
+  @property
+  def level(self):
+    def inner(level):
+      # TODO: need to define map_level
+      return f"CodecLevel={self.map_level(self.props['level'])}"
```

Comment:
Please resolve this TODO.  The map_level should be implemented in the appropriate subclass.